### DISCUSSION TITLE
Publishing to retype branch not working

### DISCUSSION THREAD

**Discussion by JoelMTaylor** [retypeapp/retype#280] [Q&A] [answered]
I have been trying to make changes to my respository. However, the changes that I have pushed have not completed. I am not quite sure how Retype works in the background, so please forgive any ignorance or error.

I have made changes to a few items. In 'Actions', the workflows are 'in progress' - and they have been doing so for a while. After leaving them to continue overnight, they timed out after about four hours. I then refreshed the workflow to continue working and they have been 'in progress' for four hours or more.
![image](https://user-images.githubusercontent.com/97919641/156051775-d6a7d25b-ff12-4783-a530-f7e08c83424b.png)
The individual workflow is: 
![image](https://user-images.githubusercontent.com/97919641/156051858-e20151a0-dd56-4d97-b5e1-f6614d6892d7.png)

I set up the workflow > retype.yml file at the beginning and didn't make any changes to that afterward. There were not issues until now. I only guess it is related to Retype as I have no experience with Github or Retype.

Please let me know what other information I can provide to assist in fixing this issue.

Thank you.

**Comment by geoffreymcgill** [comment]
Hi @JoelMTaylor. Thanks for contacting us.

I see your repo and we will investigate right away. 

The processes should only take 30-45 seconds (or close to that) to run, so anything running beyond that can be killed. Something is causing those actions to hang. We will try to figure out why.

**Reply by geoffreymcgill** [reply]
A few quick things I noticed:

1. Kill all the running processes. 
2. The [`integrations`](https://github.com/JoelMTaylor/Documentation/blob/main/retype.yml#L47-L53) section should be removed as these are not associated to your website and will not collect data properly. 
3. The use of `<img>` and `[!embed]` syntax is not going to work in your `retype.yml`. You should be able to simplify to point to just the file paths and Retype will handle the rest. 
```yml
  logo: <img scr="/static/ccm-logo.svg" width="100" height="100" /img>
  logoDark: [!embed el="embed" width="100"](/static/ccm-logo.svg)
```
Should be revised to something like:
```yml
  logo: /static/ccm-logo.svg
  logoDark: /static/ccm-logo.svg
```

All the `<img>` value would be the same. For instance,

https://github.com/JoelMTaylor/Documentation/blob/main/retype.yml#L23

```
// Existing
icon: <img scr="/static/Teams.png" width="100" height="100" /img>

// Revised
icon: /static/Teams.png
```
I can send a pull-request with the changes if you want. 

We are still investigating why the actions are hanging and attempting to reproduce.

**Reply by geoffreymcgill** [reply]
Okay, we think you have burned through all your available GitHub Action build credits. It's based on amount of time used. 

https://github.com/retypeapp/retype/discussions/280#discussioncomment-2268543

Although, it's unclear why this happened. We could be wrong too; still investigating.

**Reply by JoelMTaylor** [reply]
Yes, it does seem to be the Action credits. 

However, I will make the changes you suggested. 
The reason for the logo code was because the image wasn't showing with the standard link (the same you suggested). I was investigating other options to get them to successfully show.

Thank you.

**Reply by geoffreymcgill** [reply]
Using `logo: /static/ccm-logo.svg` should work. The path looks correct. 

You can also remove the `logoDark` config if it's not different than `logo`. Currently, they are configured with the same image paths, so only `logo` is required. Dark mode will just use the same image as light mode.

**Comment by fabriciomurta** [comment, marked as answer]
Have you looked up your [GitHub Actions workflow time usage](https://docs.github.com/en/billing/managing-billing-for-github-actions/viewing-your-github-actions-usage)? Maybe you used all your free time and workflows are being queued until you either pay for more "time" or the time limit counter renews? Notice this time is shared with other repositories you may have and, if you got some macos workflows, they consume 10 times over windows or linux ones; like, 10 minutes billed per minute ran.

**Reply by JoelMTaylor** [reply]
Yes, you are correct. I didn't know this was a thing. Thank you for your assistance.

**Reply by fabriciomurta** [reply]
By cloning your repo I noticed retype was hanging even if I ran locally! The code you added to `logo` and `logoDark` actually broke it. :)

We'll log an issue to at least not allow Retype to lock up when arbitrary code is provided to those settings (and possibly other), but that'd be a tough one because actually `logoDark: [!embed el="embed" width="100"](/static/ccm-logo.svg)` is invalid syntax even in the YAML standard. And the logo and logoDark fields [don't really accept anything other than path or a "SVG path"](https://retype.com/configuration/project/#logo). But again, it should not *hang Retype* due to that. It should report a syntax error and return with error status.

**Comment by fabriciomurta** [comment]
The actual problem here after investigating was that having `[]()` as a value (in this case it was `[!embed el="embed" width="100"](/static/ccm-logo.svg)`) to most (if not any) field in Retype configuration file (**retype.yml**) would cause an exception to be thrown and Retype hung instead of handling it appropriately.

The good news though is that some days ago we internally spotted that issue and already fixed in our forthcoming v2 release. So, for now, just play safe with YAML Syntax in the config file and no hang-ups should happen.

**Reply by fabriciomurta** [reply]
And, as for the issue your SVG logo don't display at all, it is because they do not have [intrinsic sizing](https://www.w3.org/Graphics/SVG/WG/wiki/Intrinsic_Sizing) set up. All you need to do to fix that is, open your .svg file with your favorite text editor and add `width="541" height="206"` to its first tag (`<svg ...>`). You can, of course, change the values to what fits best for you.

Retype currently does not keep a static "box" for logos, so a size is required in the SVG file.
